tests/test_common.c

```diff
@@ -295,3 +300,91 @@ test_helper_create_state_file (const char *name,
 
 	return true;
 }
+
+/**
+ * Run a VM that can be used to test qmp
+ *
+ * \param[out] socket_path qmp socket path
+ *
+ * \return qemu's pid on success, else -1
+ */
+pid_t run_qmp_vm(char **socket_path) {
+	struct stat st;
+	pid_t pid;
+	int err_pipe[2] = { -1, -1 };
+	char tmpdir[] = "/tmp/.qmpXXXXXX";
+	char err;
+	ssize_t bytes;
+	pid_t ret = -1;
+
+	if (! mkdtemp (tmpdir)) {
+		fprintf (stderr, "failed to create tmp directory");
+		goto out;
+	}
+
+	*socket_path = g_strdup_printf ("%s/hypervisor.sock", tmpdir);
+
+	if (pipe2(err_pipe, O_CLOEXEC) < 0) {
+		fprintf (stderr, "failed to create pipes");
+		return -1;
+	}
+
+	pid = fork ();
+	if (pid < 0) {
+		fprintf (stderr, "failed to fork parent");
+		return -1;
+	} else if (! pid) {
+		/* child */
+		int i = 0;
+		close (err_pipe[0]);
+
+		char **argv = calloc(12+1, sizeof(char *));
```

Comment:
I know we have this sort of logic in other places, but from the maintenance perspective, it'll be easy to overstep this buffer. How about adding a `#define QEMU_ARGS 12`, then adding a `g_assert (i <= QEMU_ARGS)` after all the args have been set in `argv` to catch an overrun?